lib/sisimai/bite/email/imailserver.rb

```diff
@@ -0,0 +1,181 @@
+      #                                   part or nil if it failed to parse or
+      #                                   the arguments are missing
+      def scan(mhead, mbody)
+        return nil unless mhead
+        return nil unless mbody
+
+        match  = 0
+        match += 1 if mhead['subject'] =~ Re0[:subject]
+        match += 1 if mhead['x-mailer'] && mhead['x-mailer'] =~ Re0[:'x-mailer']
+        return nil if match.zero?
+
+        dscontents = [Sisimai::Bite.DELIVERYSTATUS]
+        hasdivided = mbody.split("\n")
+        rfc822list = []     # (Array) Each line in message/rfc822 part string
+        blanklines = 0      # (Integer) The number of blank lines
+        readcursor = 0      # (Integer) Points the current cursor position
+        recipients = 0      # (Integer) The number of 'Final-Recipient' header
+        v = nil
+
+        hasdivided.each do |e|
+          if readcursor.zero?
+            # Beginning of the bounce message or delivery status part
+            if e =~ Re1[:begin]
+              readcursor |= Indicators[:deliverystatus]
+              next
+            end
+          end
+
+          if readcursor & Indicators[:'message-rfc822'] == 0
+            # Beginning of the original message part
+            if e =~ Re1[:rfc822]
+              readcursor |= Indicators[:'message-rfc822']
+              next
+            end
+          end
+
+          if readcursor & Indicators[:'message-rfc822'] > 0
+            # After "message/rfc822"
+            if e.empty?
+              blanklines += 1
+              break if blanklines > 1
+              next
+            end
+            rfc822list << e
+
+          else
+            # Before "message/rfc822"
+            break if readcursor & Indicators[:'message-rfc822'] > 0
+
+            # Unknown user: kijitora@example.com
+            #
+            # Original message follows.
+            v = dscontents[-1]
+
+            if cv = e.match(/\A(.+)[ ](.+)[:][ \t]*([^ ]+[@][^ ]+)/)
+              # Unknown user: kijitora@example.com
+              if v['recipient']
+                # There are multiple recipient addresses in the message body.
+                dscontents << Sisimai::Bite.DELIVERYSTATUS
+                v = dscontents[-1]
+              end
+              v['diagnosis'] = cv[1] + ' ' + cv[2]
+              v['recipient'] = cv[3]
+              recipients += 1
+
+            elsif cv = e.match(/\Aundeliverable[ ]+to[ ]+(.+)\z/)
+              # undeliverable to kijitora@example.com
+              if v['recipient']
+                # There are multiple recipient addresses in the message body.
+                dscontents << Sisimai::Bite.DELIVERYSTATUS
+                v = dscontents[-1]
+              end
+              v['recipient'] = cv[1]
+              recipients += 1
+
+            else
+              # Other error message text
+              v['alterrors']  += ' ' + e if v['alterrors']
```

Comment:
__[rubocop]__
* Layout/SpaceAroundOperators: Operator `+=` should be surrounded by a single space. [:ref](https://github.com/bbatsov/ruby-style-guide#spaces-operators)

<!-- Comment by [SideCI](https://sideci.com) -->